src/Common/src/TypeSystem/CodeGen/TypeDesc.CodeGen.cs

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Internal.TypeSystem
+{
+    // Extensions to TypeDesc suitable for code generation purposes
+    partial class TypeDesc
+    {
+        /// <summary>
+        /// Validates that a type can be fully loaded. Throws an exception if a failure occurs.
+        /// </summary>
+        public virtual void ValidateCanLoad()
```

Comment:
> Are you suggesting we should expose `EnsureInstanceLayout`, `EnsureStaticLayout`

This sounds good to me. It has clearly defined semantics.

> `EnsureRuntimeInterfaces`, etc. on TypeDesc instead

You will need to ensure that all interfaces are good given the current EEType structure - so you will need to ask for the actual list of interfaces.

> RyuJIT only asks for the R2R helper.

We should do the validation of the EETypeNode when we are creating the helper - kind of what you have there today.